spec/index.bs

```diff
@@ -4168,6 +4173,113 @@ enum GPUStoreOp {
+
+The render pass encoder has several methods which affect how draw commands are rasterized to the
+attachments used by this encoder.
+
+### <dfn method for=GPURenderPassEncoder>setViewport(x, y, width, height, minDepth, maxDepth)</dfn> ### {#GPURenderPassEncoder-setViewport}
+
+<div algorithm="GPURenderPassEncoder.setViewport">
+    <strong>|this|:</strong> of type {{GPURenderPassEncoder}}.
+
+    **Arguments:**
+        - `float` x - Left side of the viewport in pixels
+        - `float` y - Top side of the viewport in pixels
+        - `float` |width| of the viewport in pixels
+        - `float` |height| of the viewport in pixels
+        - `float` |minDepth| - Minimum depth value of the viewport
+        - `float` |maxDepth| - Maximum depth value of the viewport. May be less than or equal to
+            |minDepth|.
+
+    **Returns:** void
+
+    Sets the {{GPURenderPassEncoder/[[viewport]]}} used during the rasterization stage to linearly
+    map from normalized device coordinates to viewport coordinates.
+
+    On the [=Device timeline=], the following steps occur:
+
+        - If the [$GPURenderPassEncoder.setViewport/setViewport Valid Usage$] rules are met:
+
+            - Set |this|.{{GPURenderPassEncoder/[[viewport]]}} to the extents passed to the method.
+
+    <div class=validusage dfn-for=GPURenderPassEncoder.setViewport>
+        <dfn abstract-op>setViewport Valid Usage</dfn>
+
+        1. |width| must be greater than `0`.
+        1. |height| must be greater than `0`.
+        1. |minDepth| must be greater than or equal to `0.0` and less than or equal to `1.0`.
+        1. |maxDepth| must be greater than or equal to `0.0` and less than or equal to `1.0`.
+
+    </div>
+
+    Issue: Allowed for GPUs to use fixed point or rounded viewport coordinates
+
+</div>
+
+### <dfn method for=GPURenderPassEncoder>setScissorRect(x, y, width, height)</dfn> ### {#GPURenderPassEncoder-setScissorRect}
+
+<div algorithm="GPURenderPassEncoder.setScissorRect">
+    <strong>|this|:</strong> of type {{GPURenderPassEncoder}}.
+
+    **Arguments:**
+        - {{GPUIntegerCoordinate}} x - Left side of the scissor rectangle in pixels
+        - {{GPUIntegerCoordinate}} y - Top side of the scissor rectangle in pixels
+        - {{GPUIntegerCoordinate}} |width| of the scissor rectangle in pixels
+        - {{GPUIntegerCoordinate}} |height| of the scissor rectangle in pixels
+
+    **Returns:** void
+
+    Sets the {{GPURenderPassEncoder/[[scissor_rectangle]]}} used during the rasterization stage.
+    After transformation into viewport coordinates any fragments which fall outside the scissor
+    rectangle will be discarded.
+
+    On the [=Device timeline=], the following steps occur:
+
+        - If the [$GPURenderPassEncoder.setScissorRect/setScissorRect Valid Usage$] rules are met:
+
+            - Set |this|.{{GPURenderPassEncoder/[[scissor_rectangle]]}} to the extents passed to the method.
```

Comment:
nit: IMO clearer to just write out
```suggestion
            - Set |this|.{{GPURenderPassEncoder/[[scissor_rectangle]]}} to the extents |x|, |y|, |width|, and |height|.
```

Reply:
this change doesn't seem to be in?

Reply:
Oh, weird. I definitely merged it, so I must have accidentally stomped on it while rebasing. It's back in now!